ios/FluentUI.Demo/FluentUI.Demo/Demos/CommandBarDemoController.swift

```diff
@@ -0,0 +1,192 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import FluentUI
+import UIKit
+
+class CommandBarDemoController: DemoController {
+    enum Command {
+        case add
+        case mention
+        case calendar
+
+        case textBold
+        case textItalic
+        case textUnderline
+        case textStrikethrough
+
+        case arrowUndo
+        case arrowRedo
+
+        case copy
+        case delete
+
+        case checklist
+        case bulletList
+        case numberList
+        case link
+
+        case keyboard
+
+        var iconImage: UIImage? {
+            switch self {
+            case .add:
+                return UIImage(named: "add24Regular")
+            case .mention:
+                return UIImage(named: "mention24Regular")
+            case .calendar:
+                return UIImage(named: "calendar24Regular")
+            case .textBold:
+                return UIImage(named: "textBold24Regular")
+            case .textItalic:
+                return UIImage(named: "textItalic24Regular")
+            case .textUnderline:
+                return UIImage(named: "textUnderline24Regular")
+            case .textStrikethrough:
+                return UIImage(named: "textStrikethrough24Regular")
+            case .arrowUndo:
+                return UIImage(named: "arrowUndo24Regular")
+            case .arrowRedo:
+                return UIImage(named: "arrowRedo24Filled")
+            case .copy:
+                return UIImage(named: "copy24Regular")
+            case .delete:
+                return UIImage(named: "delete24Regular")
+            case .checklist:
+                return UIImage(named: "textChecklistListLtr24Regular")
+            case .bulletList:
+                return UIImage(named: "textBulletList24Regular")
+            case .numberList:
+                return UIImage(named: "textNumberListLtr24Regular")
+            case .link:
+                return UIImage(named: "link24Regular")
+            case .keyboard:
+                return UIImage(named: "keyboardDock24Regular")
+            }
+        }
+
+        var isPersistSelection: Bool {
+            switch self {
+            case .add, .mention, .calendar, .arrowUndo, .arrowRedo, .copy, .delete, .link, .keyboard:
+                return false
+            case .textBold, .textItalic, .textUnderline, .textStrikethrough, .checklist, .bulletList, .numberList:
+                return true
+            }
+        }
+    }
+
+    class Item: CommandBarItem {
```

Comment:
Is it expected that the consumer should have to sub-class, or that is optional (there seems to be some problem with the ObjC headers which are being exposed so it is hard to see the intended usage)

Reply:
It is optional. But it would be hard to identify which button is tapped without subclassing and adding a property specifically for this purpose.

Reply:
Using a string as identifier is not a friendly API for Swift. I originally intended to use generic feature of Swift, to enable providing a custom enum type to identify buttons. But that would make CCB completely not compatible with ObjC.

Reply:
Okay, thanks, for my purposes I can look at using a Swift subclass, but the general design is still an issue if you are determined to use ObjC. The block option may be better, depending on what the Fluent team say